src/app/components/artist_details/artist_details_model.rs

```diff
@@ -123,11 +123,27 @@ impl PlaylistModel for ArtistDetailsModel {
 
         let menu = gio::Menu::new();
         menu.append(Some(&*labels::VIEW_ALBUM), Some("song.view_album"));
-        for artist in song.artists.iter().filter(|a| self.id != a.id) {
-            menu.append(
-                Some(&labels::more_from_label(&artist.name)),
-                Some(&format!("song.view_artist_{}", artist.id)),
-            );
+
+        let artists_iter = song.artists.iter();
+
+        if artists_iter.len() > 1 {
+            let submenu = gio::Menu::new();
+
+            for artist in artists_iter {
+                submenu.append(
+                    Some(&artist.name),
+                    Some(&format!("song.view_artist_{}", artist.id)),
+                );
+            }
+
+            menu.append_submenu(Some("More from"), &submenu);
```

Comment:
these strings wont be translated -- you have to add a gettext call here, or I guess we could cheat and have `labels::more_from_label("")` here :see_no_evil: